src/internal/connector/exchange/service_iterators.go

```diff
@@ -43,6 +47,49 @@ func getAdditionalDataString(
 	return *value
 }
 
+// makeMetadataCollection creates a metadata collection that has a file
+// containing all the delta tokens in tokens. Returns nil if the map does not
+// have any entries.
+//
+// TODO(ashmrtn): Expand this/break it out into multiple functions so that we
```

Comment:
There's very little here that's either delta token or even Exchange specific.  I think this would be better moved into `metadata_collection.go`.  Then have it take a path.service and a file name, and it works for everything.

Reply:
if we just have it take a file name then we won't be able to have multiple separate metadata files because calling this function 2x will result in 2 separate collections with the same path but different items. I think either the function needs to assume it's getting some fixed input of data to serialize into well-known files or it should stay local to whatever code is using it to create files in the metadata collection

Reply:
Mmmkay, keep it as is for now.  We can watch for refactoring/consolidation in the future as needed.